### PR TITLE
chore: strict tsconfig

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Following what AI Lab did, just upping our backend tsconfig settings.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Part of #247.

### How to test this PR?

Just adds additional checks to PR build.

### CODE REVIEW

File: packages/backend/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,10 @@
 {
   "compilerOptions": {
     "strictNullChecks": true,
+    "strictBindCallApply": true,
+    "alwaysStrict": true,
+    "strictFunctionTypes": true,
+    "noImplicitThis": true,
```
Comment (benoitf) [inline packages/backend/tsconfig.json:7]: if possible go fo `"strict": true,`

Comment (deboer-tim) [reply packages/backend/tsconfig.json]: Done.

**Review by benoitf — APPROVED** [review]

File: packages/backend/src/build-disk-image.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export async function buildDiskImage(build: BootcBuildInfo, history: History): P
 
   // VALIDATION CHECKS
   for (const { field, message } of requiredFields) {
-    if (!build[field]) {
+    if (!build[field as keyof BootcBuildInfo]) {
```
Comment (benoitf) [inline packages/backend/src/build-disk-image.ts:47]: for this one it's look tricky

Comment (deboer-tim) [reply packages/backend/src/build-disk-image.ts:47]: Yeah. I like that it actually says what it's doing though, and other options I tried were more verbose, e.g.:
`if (!(build as {[key: string]: any})[field]) {`
or:
`if (!(build as Record<string, any>)[field]) {`
or:
```
const obj: Record<string, any> = build;
if (!obj[field]) {
```

Prefer any of these or a different one?

Comment (deboer-tim) [reply packages/backend/src/build-disk-image.ts:47]: Merging, feel free to ping if you'd like me to change it :).

Comment (benoitf) [reply packages/backend/src/build-disk-image.ts:47]: ah yes no pb, I was hoping that we don't need to use any cast there but maybe it's not possible but anyway it's was not preventing any merge 👍

**Pull request merged**